### DISCUSSION TITLE
Avez-vous besoin de NgRx ? | Bonjour Angular

### DISCUSSION THREAD

**Discussion by giscus** [bonjour-angular/apprendre#31] [General]
# Avez-vous besoin de NgRx ? | Bonjour Angular

Avez-vous besoin de NgRx ?

https://apprendre.bonjour-angular.com/articles/avez-vous-besoin-de-ngrx/

**Comment by memory660** [comment]
je dev sur angular depuis 10 ans et je déteste NgRx qui pour moi vient du monde de react et il aurait du rester la bas.
oui, créer 3 fichiers cela me saoule au plus haut point.

pour ton exemple de loadTodos, je ferais comme ça:

on a une fonction  loadTodos(): Observable<Todo[]);
si je dois faire 3 actions suite au loadTodos, je créais une nouvelle fonction :
```
loadTodosAndComputeActions(): Observable<Todos[]> {
     return this.loadTodos().pipe(
         switchMap((todos: Todos[]) => { 
                  ....  action 1....
                  ....  action 2....
                  ....  action 3....
                 return todos;
          )
     )
}
// switchmap  ou avec 3 tap
```
separation of concerns n'est pas brisée puisque c'est loadTodosAndComputeActions  et non loadTodos
donc il fait bien ce qu'il prétend être

c'est rapide à modifier, fiable, claire....


si je veux faire les 3 actions ---> loadTodosAndComputeActions()
sans faire les 3 actions :  loadTodos()


par contre si tu as definis un effect sur loadTodos
les actions seront toujours fait et si dans certains cas on ne veut pas ?
donc faut encore faire des choses dans les 3 fichiers
